### PR TITLE
continuous integration setup

### PR DESCRIPTION
**Pull request by Aijeyomah**
#### What does this PR do?
setup continuous in app 
 
#### Description of Task to be completed?
- Setup continuous integration with travis ci and test coverage analysis with coveralls.
- Write unit test.

#### How should this be manually tested?
- Clone the repo and cd into the project folder
- Checkout to ch-basic-config branch and run npm install.
- Run npm test to see results.

### CODE REVIEW

File: .nyc_output/1cb36342-4ad8-4fef-911f-825f70204037.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (daylay92) [inline .nyc_output/1cb36342-4ad8-4fef-911f-825f70204037.json]: Please remove the coverage and .nyc_output folders from the staging area. To do this:
-  run `git rm --cached -r. <name-of-the-folders-should-replace this>` e.g `git rm --cached -r coverage`
- check for changes with `git status`
- add the folders to the `.gitignore` file.
- make a commit
- push your changes

File: README.md [hunk 2]
```diff
@@ -43,4 +43,7 @@ $ cd book-club
 $ npm install
 ```
 
-
```
Comment (daylay92) [inline README.md:46]: Move the badges to the top of your file. Also, you haven't properly configured codeclimate yet. Look into the travis build log for possible errors and rectify them.

File: package.json [hunk 3]
```diff
@@ -4,9 +4,12 @@
   "description": "A platform where people upload add books ",
   "main": "index.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1",
+    "test": "nyc --reporter=html --reporter=text-summary mocha -r esm test/**/*.test.js",
+    "test-watch": "nodemon --exec npm test",
     "lint": "eslint --fix --ext .js .",
-    "start": "node -r esm ."
+    "start": "node -r esm .",
+    "coveralls": "nyc npm run test-build && nyc report --reporter=text-lcov | coveralls",
```
Comment (daylay92) [inline package.json]: Why do you have a test-build on this line?. I don't see a test build in amongst your scripts?

File: package.json [hunk 4]
```diff
@@ -4,9 +4,12 @@
   "description": "A platform where people upload add books ",
   "main": "index.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1",
+    "test": "nyc --reporter=html --reporter=text-summary mocha -r esm test/**/*.test.js",
```
Comment (daylay92) [inline package.json]: Add this line as a flag here `--reporter=text-lcov` this might be why your codeclimate test coverage isn't working.

**Review by daylay92 — COMMENTED** [review]
@Aijeyomah Kindly improve on the use of markdown. I made some corrections to your PR. Also, resolve all the issues raised before I approve this PR. Good Job so far.

File: README.md (hunk 2 above)
Comment (Aijeyomah) [reply README.md:46]: Screenshot 2020-08-27 at 15 25 52

This is the screenshot of my travis log

Comment (Aijeyomah) [reply README.md:46]: The build log is empty.

Comment (daylay92) [reply README.md:46]: Okay, I'd have a look.

Comment (daylay92) [reply README.md:46]: Is this a public or a private repo? Can you switch over to [travis-ci](https://travis-ci.org). Add the project there and trigger a build.

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,32 @@
+language: node_js
+
+node_js:
+  - "v12.18.1"
+  install:
```
Comment (daylay92) [inline .travis.yml]: you haven't properly indented this file.

File: .travis.yml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+     - CC_TEST_REPORTER_ID= 0af239fa6a55520e8cefc4dcde153ff7293ad506c101d9c14cc8b8eaa10dadc7
+
+before_script:
+  - psql -c "CREATE DATABASE testing_db;" -U postgres
+  - curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > ./cc-test-reporter
+  - chmod +x ./cc-test-reporter
+  - ./cc-test-reporter before-build
+
+script:
+  - npm run migrations
```
Comment (daylay92) [inline .travis.yml]: Why do you have this script, it doesn't exist in your package.json file.

File: README.md [hunk 7]
```diff
@@ -1,3 +1,8 @@
+[![Coverage Status](https://coveralls.io/repos/github/Aijeyomah/Book-club/badge.svg?branch=develop)](https://coveralls.io/github/Aijeyomah/Book-club?branch=develop)
+[![CodeFactor](https://www.codefactor.io/repository/github/aijeyomah/book-club/badge)](https://www.codefactor.io/repository/github/aijeyomah/book-club)
+[![Maintainability](https://api.codeclimate.com/v1/badges/c3afcd24a7704ac72efc/maintainability)](https://codeclimate.com/github/Aijeyomah/Book-club/maintainability)
+[![Test Coverage](https://api.codeclimate.com/v1/badges/c3afcd24a7704ac72efc/test_coverage)](https://codeclimate.com/github/Aijeyomah/Book-club/test_coverage)
+
```
Comment (daylay92) [inline README.md:6]: Add the build badge on travis ci to the this file.

**Pull request merged**